py/asmrv32.h

```diff
@@ -151,23 +151,40 @@ void asm_rv32_end_pass(asm_rv32_t *state);
     ((op & 0b1111111) | ((rd & 0b11111) << 7) |                                  \
     (imm & 0b11111111111111111111000000000000))
 
+#define RV32_ENCODE_TYPE_CB(op, ft3, rs, imm) \
+    ((op & 0b11) | ((ft3 & 0b111) << 13) | ((rs & 0b111) << 7) | \
```

Comment:
I only just realised now that you are using binary literals.  AFAIK this is not in the C standard, at least not in C99 which we aim to support.

I think it's only working because gcc (and other compilers) has an extension to support binary literals.

I think we need to use hex literals instead.

Reply:
Fair enough, but consider that all known compilers that can build code for RISC-V (GCC, LLVM, IAR, and Segger) have that turned on by default.

Turning all those binary literal into their hex counterparts is not a problem, but I wonder what's the best course of action.  Merge this PR after review and then I submit another one to turn binary literals into hex, or to hold this off for a bit until I send the conversion PR and it gets merged, then I update this one.

I'd rather send a conversion PR that covers both this and the previous one, but if you prefer the other option (or something else entirely) that'd be fine for me too.

Reply:
I would prefer to stay away from binary literals (even if they are very neat in situations like this).  And that's because it would be too easy for them to accidentally be copied to code that is not purely for RV32.  We need consistency and it's hard to say "binary literals can only be used for RV32 code".

Furthermore, `mpy-cross` actually builds the RV32 native emitter so it can target all archs.  And that builds with an arbitrary compiler, and can run on any target machine (even in the browser through WASM).  So there may be an issue building `mpy-cross` if someone uses a compiler that doesn't support binary literals.

Reply:
> I'd rather send a conversion PR that covers both this and the previous one,

That's OK by me.